core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -480,6 +486,11 @@ public Builder withObjectMapper(ObjectMapper objectMapper) {
       return this;
     }
 
+    public Builder withTlsSocketStrategy(TlsSocketStrategy socketStrategy) {
+      this.tlsSocketStrategy = socketStrategy;
```

Comment:
we should make sure in the build method that this isn't null. Also we should have some tests when this is null or configured to a real value